code/hsec-tools/src/Security/OSV.hs

```diff
@@ -0,0 +1,223 @@
+-- | This module contains the OSV datatype and its ToJSON instance.
+-- The module was initialized with http://json-to-haskell.chrispenner.ca/
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Security.OSV where
+
+import Data.Aeson (ToJSON(..), FromJSON(..), Value(..), (.:), (.=), object)
+import Data.Aeson.Types (prependFailure, typeMismatch)
+import Data.Text (Text)
+
+data Affected = Affected
+  { affectedRanges :: [Ranges]
+  , affectedPackage :: Package
+  , affectedEcosystemSpecific :: EcosystemSpecific
+  , affectedDatabaseSpecific :: DatabaseSpecific
+  } deriving (Show, Eq, Ord)
+
+data Affects = Affects
+  { affectsOs :: [Value]
+  , affectsArch :: [Value]
+  , affectsFunctions :: [Value]
+  } deriving (Show, Eq, Ord)
+
+data DatabaseSpecific = DatabaseSpecific
+  { databaseSpecificCvss :: Maybe Value
+  , databaseSpecificCategories :: [Value]
+  , databaseSpecificInformational :: Text
+  } deriving (Show, Eq, Ord)
+
+data EcosystemSpecific = EcosystemSpecific
+  { ecosystemSpecificAffects :: Affects
+  } deriving (Show, Eq, Ord)
+
+data Events = Events
+  { eventsIntroduced :: Text
```

Comment:
Should we also include the other 3 types of events? Even if we don't use them right now?

All of these would be optional.

https://ossf.github.io/osv-schema/#affectedrangesevents-fields